sdk/storage/Azure.Storage.DataMovement/src/ServiceToServiceTransferJob.cs

```diff
@@ -171,31 +171,49 @@ private async IAsyncEnumerable<JobPartInternal> GetStorageResourcesAsync()
                 }
 
                 StorageResource current = enumerator.Current;
-                if (!existingSources.Contains(current.Uri))
+
+                if (current.IsContainer)
                 {
+                    // Create sub-container
                     string containerUriPath = _sourceResourceContainer.Uri.GetPath();
-                    string sourceName = string.IsNullOrEmpty(containerUriPath)
+                    string subContainerPath = string.IsNullOrEmpty(containerUriPath)
                         ? current.Uri.GetPath()
                         : current.Uri.GetPath().Substring(containerUriPath.Length + 1);
-
-                    ServiceToServiceJobPart part;
-                    try
-                    {
-                        part = await ServiceToServiceJobPart.CreateJobPartAsync(
-                            job: this,
-                            partNumber: partNumber,
-                            sourceResource: (StorageResourceItem)current,
-                            destinationResource: _destinationResourceContainer.GetStorageResourceReference(sourceName))
-                            .ConfigureAwait(false);
-                        AppendJobPart(part);
-                    }
-                    catch (Exception ex)
+                    StorageResourceContainer subContainer =
+                        _destinationResourceContainer.GetChildStorageResourceContainer(subContainerPath);
```

Comment:
Why do we get a sub directory resource here? Shouldn't we just create the `current` container?

Reply:
Because we can't call `Create` on the `StorageResource current`. `Create` currently exists only in the `StorageResourceContainer` class.

I did consider adding it to the general `StorageResource` abstract class, but then `Create` would only be called for storage resources that are containers, which would only be the `StorageResourceContainer` objects.

Reply:
So you are checking `IsContainer` above so could you cast current to a container and then call create?